File: app/src/main/java/com/ducktapedapps/updoot/utils/SingleLiveEvent.java
package com.ducktapedapps.updoot.utils;

//event wrapper used to consume events such as Toast messages only once, not using this will cause a Toast message on config change
public class SingleLiveEvent<T> {
    private T content;
    private boolean isHandled;

    public SingleLiveEvent(T content) {
        this.content = content;
        this.isHandled = false;
    }

    public T peekContent() {
        return this.content;
    }

    public T getContentIfNotHandled() {
        if (!isHandled) {
            isHandled = true;
            return content;
        } else {
            return null;
        }
    }
}

File: app/src/main/java/com/ducktapedapps/updoot/utils/SingleLiveEvent.kt
package com.ducktapedapps.updoot.utils

//event wrapper used to consume events such as Toast messages only once, not using this will cause the Toast message to be displayed again on config change
class SingleLiveEvent<T>(private val content: T) {
    private var isHandled: Boolean = false

    val contentIfNotHandled: T?
        get() {
            return if (!isHandled) {
                isHandled = true
                content
            } else {
                null
            }
        }

    init {
        this.isHandled = false
    }

    fun peekContent(): T {
        return this.content
    }
}
